viz_scripts/scaffolding.py

```diff
@@ -192,6 +194,25 @@ def translate_labels(labels):
         dic_mapping = translate_labels(dynamic_labels[label_type])
         return dic_mapping
 
+# Function: Maps "MODE", "PURPOSE", and "REPLACED_MODE" to colors.
+# Input: dynamic_labels, dic_re, and dic_pur
+# Output: Map for color with mode and purpose
+def mapping_color_labels(dynamic_labels, dic_re, dic_pur):
+    if len(dynamic_labels) > 0:
+        mode_values = list(mapping_labels(dynamic_labels, "MODE").values()) if "MODE" in dynamic_labels else []
```

Comment:
future cleanup: I think this will not work if we have two keys with the same value in the dynamic labels - e.g.

```
   "Gas car, drove alone": "drove_alone",
   "E-car, drove alone", "drove_alone"
```

That is, of course, incorrect input, but I am not sure we handle it gracefully. Failing silently is not gracefully.